tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -363,61 +363,67 @@ var _ = Describe("odo devfile push command tests", func() {
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-valid-events.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-
-			output := helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
-			helper.MatchAllInOutput(output, []string{"PreStart commands have been added to the component"})
-
-			firstPushPodName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
-
-			firstPushInitContainers := commonVar.CliRunner.GetPodInitContainers(cmpName, commonVar.Project)
-			// 3 preStart events + 1 supervisord init containers
-			Expect(len(firstPushInitContainers)).To(Equal(4))
-			helper.MatchAllInOutput(strings.Join(firstPushInitContainers, ","), expectedInitContainers)
-
-			// Need to force so build and run get triggered again with the component already created.
-			output = helper.CmdShouldPass("odo", "push", "--project", commonVar.Project, "-f")
-			helper.MatchAllInOutput(output, []string{"PreStart commands have been added to the component"})
-
-			secondPushPodName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
-
-			secondPushInitContainers := commonVar.CliRunner.GetPodInitContainers(cmpName, commonVar.Project)
-			// 3 preStart events + 1 supervisord init containers
-			Expect(len(secondPushInitContainers)).To(Equal(4))
-			helper.MatchAllInOutput(strings.Join(secondPushInitContainers, ","), expectedInitContainers)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-preStart.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
 
-			Expect(firstPushPodName).To(Equal(secondPushPodName))
-			Expect(firstPushInitContainers).To(Equal(secondPushInitContainers))
-
-			var statErr error
-			commonVar.CliRunner.CheckCmdOpInRemoteDevfilePod(
-				firstPushPodName,
-				"runtime",
-				commonVar.Project,
-				[]string{"cat", "/projects/test.txt"},
-				func(cmdOp string, err error) bool {
-					if err != nil {
-						statErr = err
-					} else if cmdOp == "" {
-						statErr = fmt.Errorf("prestart event action error, expected: hello test2\nhello test2\nhello test\n, got empty string")
-					} else {
-						fileContents := strings.Split(cmdOp, "\n")
-						if len(fileContents)-1 != 3 {
-							statErr = fmt.Errorf("prestart event action count error, expected: 3 strings, got %d strings: %s", len(fileContents), strings.Join(fileContents, ","))
-						} else if cmdOp != "hello test2\nhello test2\nhello test\n" {
-							statErr = fmt.Errorf("prestart event action error, expected: hello test2\nhello test2\nhello test\n, got: %s", cmdOp)
+			output := helper.CmdShouldFail("odo", "push", "--project", commonVar.Project)
+			// This is expected to fail for now.
+			// see https://github.com/openshift/odo/issues/4187 for more info
+			helper.MatchAllInOutput(output, []string{"\"preStart\" is not supported in odo"})
+
+			/*
```

Comment:
Any reason why this piece of code is commented instead of being removed?

Reply:
My reasoning was that disabling `preStart` is a temporary measure. Once it is clear how it should be implemented this might be useful again.

Reply:
Makes sense. :+1: